src/fu-engine.c

```diff
@@ -1412,18 +1462,25 @@ fu_engine_install (FuEngine *self,
 	/* signal to all the plugins the update has happened */
 	for (guint j = 0; j < plugins->len; j++) {
 		FuPlugin *plugin_tmp = g_ptr_array_index (plugins, j);
-		g_autoptr(GError) error_local = NULL;
-		if (!fu_plugin_runner_update_cleanup (plugin_tmp, device, &error_local)) {
+		g_autoptr(GError) error_cleanup = NULL;
+		if (!fu_plugin_runner_update_cleanup (plugin_tmp, device, &error_cleanup)) {
 			g_warning ("failed to update-cleanup: %s",
-				   error_local->message);
+				   error_cleanup->message);
 		}
 	}
 
 	/* make the UI update */
 	fu_device_set_status (device, FWUPD_STATUS_IDLE);
-	fu_device_set_modified (device, (guint64) g_get_real_time () / G_USEC_PER_SEC);
 	fu_engine_emit_device_changed (self, device);
 	fu_engine_emit_changed (self);
+
+	/* update database */
+	if (!fu_pending_set_update_state (self->pending, device,
```

Comment:
I think you'll need to introduce a separate state from success for updates that require a reboot or possibly delay setting it until the next boot on these.

Reply:
Agree 100%, although I wasn't entirely sure how to get fwupdmgr to run at the next boot (when networking is set up etc) -- maybe we just rely on nagging/gnome-software for this?

Reply:
Oh, wait that's FWUPD_UPDATE_STATE_PENDING isn't it?  I guess that needs to be set here on those that will be rebooting instead.

Reply:
Well, PENDING is strictly for updates that need to use fwupd to apply the update "offline" rather than ones that are SCHEDULED (better name welcome) and are ready to fire before the OS even boots.

Reply:
Have you looked at user agent data across LVFS lately?  It would be interesting to see the actual mix of fwupdmgr vs gnome-software.  That said nagging in gnome-software and next time they manually run fwupdmgr will still yield plenty of data.  I don't think it matters too much if it's delayed data.

A useful place to nag: you can put something in /etc/motd like Canonical does with telling you that you have X updates to do when you log into Ubuntu via SSH.

Reply:
Yup, I get a user-agent breakdown on the admin analytics page. It's 99% gnome-software all the way down. Good idea on motd, I'll investigate.